bin/dry-run.rb

```diff
@@ -402,101 +402,26 @@ def fetch_files(fetcher)
     end
   end
 rescue StandardError => e
-  error_details = handle_dependabot_error(error: e)
+  error_details = Dependabot.fetcher_error_details(e)
+  raise unless error_details
 
   puts " => handled error whilst fetching dependencies: #{error_details.fetch(:"error-type")} " \
        "#{error_details.fetch(:"error-detail")}"
 
   []
```

Comment:
Is there any reason why we cannot `raise` when there are error details? Is that because we only throw when there are unknown exceptions?

Also, rather than `puts`, is there any reason not to use the [Logger](https://github.com/dependabot/dependabot-core/blame/main/common/lib/dependabot/logger.rb) with `error` level?

Reply:
I followed the same strategy already used for fetching files. Same for using `puts`, I just did what we were already doing in other error handling blocks.